File: apps/api/src/services/userContext.service.ts
import type { HydratedDocument } from "mongoose";
import type { UpdateUserContextInput, UserContext } from "@believe-ai/shared";
import type { UserContextDocument } from "@believe-ai/server";
import { userContextRepository } from "../repositories/userContext.repository.js";

function toDto(doc: HydratedDocument<UserContextDocument>): UserContext {
  return {
    userId: doc.userId.toString(),
    aboutMe: doc.aboutMe ?? null,
    companyInfo: doc.companyInfo ?? null,
    servicesOrProducts: doc.servicesOrProducts ?? null,
    skillsAndExperience: doc.skillsAndExperience ?? null,
    achievements: doc.achievements ?? null,
    targetAudience: doc.targetAudience ?? null,
    updatedAt: doc.updatedAt.toISOString(),
  };
}

export const userContextService = {
  /** Returns null rather than a not-found error — a Believe Profile is optional. */
  async getById(userId: string): Promise<UserContext | null> {
    const doc = await userContextRepository.findByUserId(userId);
    return doc ? toDto(doc) : null;
  },

  async update(userId: string, input: UpdateUserContextInput): Promise<UserContext> {
    const doc = await userContextRepository.upsert(userId, input);
    return toDto(doc);
  },
};

File: apps/backend/services/user_context_service.py
"""Believe Profile — optional per-user context fed automatically into every
AI call that benefits from sender context."""

from __future__ import annotations

from bson import ObjectId

from models.user_context import UserContext
from repositories import user_context_repository
from schemas.user_context import UpdateUserContextInput, UserContextDto


def _to_dto(doc: UserContext) -> UserContextDto:
    return UserContextDto(
        userId=str(doc.userId),
        aboutMe=doc.aboutMe,
        companyInfo=doc.companyInfo,
        servicesOrProducts=doc.servicesOrProducts,
        skillsAndExperience=doc.skillsAndExperience,
        achievements=doc.achievements,
        targetAudience=doc.targetAudience,
        updatedAt=doc.updatedAt.isoformat(),
    )


async def get_user_context(user_id: ObjectId) -> UserContextDto | None:
    """Returns None rather than a not-found error — a Believe Profile is optional."""
    doc = await user_context_repository.find_by_user_id(user_id)
    return _to_dto(doc) if doc else None


async def update_user_context(user_id: ObjectId, updates: UpdateUserContextInput) -> UserContextDto:
    doc = await user_context_repository.upsert(user_id, updates)
    return _to_dto(doc)
